woocommerce/woocommerce-ios
# [Product Multi-selection] Hide clear selection button for product multi-selection

Closes: #8974 

## Description
This PR hides the "Clear Selection" button for both Products and Product Variations when multiple-selection is enabled, but keeps the button for Coupons. 

This is done temporarily as part of M1 as we can't add or delete multiple products from an Order in one go, and clearing the selection would currently require that all selected products are removed at the same time. This can be restored in M1.1 when we [add support for multiple additions/removals](https://github.com/woocommerce/woocommerce-ios/issues/8923).

## Changes:
- Add a new property to the `ProductSelectorView's` `Configuration` initializer, to set if the "clear selection" button should be enabled or not. This allows us to filter down and use it only on the init for `addProductToOrder` without affecting other implementations like Coupons.
- We pass this property down to `ProductVariationSelectorView` as well
- We add the property as a check inside `.renderedIf(_:)` to both Views, so we can render or not if needed.

## Testing instructions
- Switch the `.productMultiSelectionM1` feature flag to `true`
- Go to `Orders` > Tap on `+` > Tap on  `+ Add Product`
- Select a Product > See how the "Clear Selection" button does not appear at the top of the screen
- Go to `Menu` > `Coupons` (enable the feature if is disabled in Experimental Features) > Tap on `+` > Tap on `Percentage discount` > Tap on `All Products` > Confirm the "Clear Selection" button still appears at the top of the screen.

*Note: While testing you might notice the Coupons view has a `Filter` button as well, this will be added [a different PR here](https://github.com/woocommerce/woocommerce-ios/issues/8998), just to keep things scoped separately, as neither "Clear Selection" nor "Filters" was part of the original view for Product single-selection.

## Screenshots
|  Before (clear selection button)  |  After (no clear selection button) |
|---|---|
| <img width=450 src="https://user-images.githubusercontent.com/3812076/221797431-490b2fa3-d07c-4786-8c35-b482443682bf.png">  |   <img width=450 src="https://user-images.githubusercontent.com/3812076/221798172-f1d5c359-5319-4bf3-9ec0-0b046258851d.png"> |

---
